api/swagger.yml

```diff
@@ -2523,7 +2558,98 @@ paths:
           description: too many requests
         default:
           $ref: "#/components/responses/ServerError"
-
+  /auth/external/principals:
+    get: 
+      tags: 
+        - auth
+        - external
+      operationId: listExternalPrincipals
+      summary: list external principals
+      parameters: 
+        - $ref: "#/components/parameters/PaginationPrefix"
+        - $ref: "#/components/parameters/PaginationAfter"
+        - $ref: "#/components/parameters/PaginationAmount"
+      responses: 
+        200: 
+          description: external principal list
+          content: 
+            application/json: 
+              schema: 
+                $ref: "#/components/schemas/ExternalPrincipalList"
+        401: 
+          $ref: "#/components/responses/Unauthorized"
+        420: 
+          description: too many requests
+        default: 
+          $ref: "#/components/responses/ServerError"
+    post: 
+      tags: 
```

Comment:
experimental? or you want to commit to this API as part of our 1.0 guarantees?

Reply:
Thanks, added experimental tag!